applications/GeoMechanicsApplication/custom_strategies/strategies/residualbased_newton_raphson_strategy_linear_elastic_dynamic.hpp

```diff
@@ -0,0 +1,503 @@
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    /**
+     * @brief This method prints information after reach the max number of iterations
+     */
+    void MaxIterationsExceeded() override
+    {
+        KRATOS_INFO_IF("ResidualBasedNewtonRaphsonStrategyLinearElasticDynamic", BaseType::GetEchoLevel() > 0)
+            << "ATTENTION: max iterations ( " << BaseType::mMaxIterationNumber << " ) exceeded!"
+            << std::endl;
+    }
+
+    bool PerformIterationCycle(TSystemMatrixType&   rA,
+                               TSystemVectorType&   rDx,
+                               TSystemVectorType&   rb,
+                               TSystemVectorType&   rDxTot,
+                               std::vector<Vector>& rNonconvergedSolutions,
+                               unsigned int&        rIterationNumber)
+    {
+        ModelPart&                    r_model_part = BaseType::GetModelPart();
+        typename TSchemeType::Pointer p_scheme     = BaseType::GetScheme();
+        typename TBuilderAndSolverType::Pointer p_builder_and_solver = BaseType::GetBuilderAndSolver();
+        auto&       r_dof_set              = p_builder_and_solver->GetDofSet();
+        const auto& r_current_process_info = r_model_part.GetProcessInfo();
+
+        bool is_converged = false;
+        rIterationNumber++;
+        for (; rIterationNumber < BaseType::mMaxIterationNumber; rIterationNumber++) {
+            // setting the number of iteration
+            r_model_part.GetProcessInfo()[NL_ITERATION_NUMBER] = rIterationNumber;
+
+            p_scheme->InitializeNonLinIteration(r_model_part, rA, rDx, rb);
+            BaseType::mpConvergenceCriteria->InitializeNonLinearIteration(r_model_part, r_dof_set, rA, rDx, rb);
+
+            is_converged = BaseType::mpConvergenceCriteria->PreCriteria(r_model_part, r_dof_set, rA, rDx, rb);
+
+            // call the linear system solver to find the correction mDx for the
+            // it is not called if there is no system to solve
+            if (SparseSpaceType::Size(rDx) != 0) {
+                TSparseSpace::SetToZero(rDx);
+                TSparseSpace::SetToZero(rb);
+
+                p_builder_and_solver->BuildRHSAndSolve(p_scheme, r_model_part, rA, rDx, rb);
+
+            } else {
+                KRATOS_WARNING("NO DOFS") << "ATTENTION: no free DOFs!! " << std::endl;
+            }
+
+            // Debugging info
+            BaseType::EchoInfo(rIterationNumber);
+
+            // Updating the results stored in the database
+            this->UpdateSolutionStepValue(rDx, rDxTot);
+
+            // only finalize condition non linear iteration
+            block_for_each(r_model_part.Conditions(), [&r_current_process_info](Condition& r_condition) {
+                if (r_condition.IsActive()) {
+                    r_condition.FinalizeNonLinearIteration(r_current_process_info);
+                }
+            });
```

Comment:
If I'm not mistaken, this is exactly what your new scheme does...? So I think you can replace it by:
```suggestion
            p_scheme->FinalizeNonLinIteration(r_model_part, rA, rDx, rb);
```